src/ui/controllers/profile.ts

```diff
@@ -128,9 +163,78 @@ export async function updateOrganisation(request: Request, response: Response) {
 	}
 }
 
+function displayTokenPage(request: Request, response: Response, errMsg: string, value: string, organisation: string) {
+	response.send(template.render(`profile/enterToken`, request, response, {
+		error: true,
+		msg: errMsg,
+		org: value,
+		organisation,
+		originalUrl: request.body.originalUrl,
+	}))
+}
+
+export async function checkTokenValidity(request: Request, response: Response) {
+
+	const value = request.body.org
+	let organisationalUnit
+
+	try {
+		const organisationResponse: any = await registry.getWithoutHal(value)
+		organisationalUnit = {
+			code: organisationResponse.data.code,
+			name: organisationResponse.data.name,
+			paymentMethods: organisationResponse.data.paymentMethods,
+		}
+
+	} catch (error) {
+		console.log(error)
+		throw new Error(error)
+	}
+
+	const code = organisationalUnit.code
+	const domain = request.user.userName.split("@")[1]
+	const token = request.body.token
+	const accessToken = request.user.accessToken
+
+	if (!token) {
+			displayTokenPage(request, response, "Please don't leave the token blank", value, organisationalUnit.name)
+	} else {
+		const checkTokenValidResponse: any = await registry.updateToken(code, domain, token, false, accessToken)
+		if (checkTokenValidResponse === "NONE") {
```

Comment:
Use the http response code instead of a String incase the text for this ever changes.  The http response code shouldn't change in the future.

Reply:
Done